src/Surfnet/ServiceProviderDashboard/Application/Metadata/JsonGenerator.php

```diff
@@ -0,0 +1,239 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Application\Metadata;
+
+use Surfnet\ServiceProviderDashboard\Application\Metadata\JsonGenerator\ArpGenerator;
+use Surfnet\ServiceProviderDashboard\Application\Metadata\JsonGenerator\MotivationMetadataGenerator;
+use Surfnet\ServiceProviderDashboard\Application\Metadata\JsonGenerator\PrivacyQuestionsMetadataGenerator;
+use Surfnet\ServiceProviderDashboard\Application\Metadata\JsonGenerator\SpDashboardMetadataGenerator;
+use Surfnet\ServiceProviderDashboard\Application\Metadata\GeneratorInterface;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\Attribute;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\Contact;
+
+/**
+ * @SuppressWarnings(PHPMD.TooManyMethods)
+ * @SuppressWarnings(PHPMD.ElseExpression)
+ */
+class JsonGenerator implements GeneratorInterface
+{
+
+    /**
+     * @var ArpGenerator
+     */
+    private $arpMetadataGenerator;
+
+    /**
+     * @var PrivacyQuestionsMetadataGenerator
+     */
+    private $privacyQuestionsMetadataGenerator;
+
+    /**
+     * @var MotivationMetadataGenerator
+     */
+    private $motivationMetadataGenerator;
+
+    /**
+     * @var SpDashboardMetadataGenerator
+     */
+    private $spDashboardMetadataGenerator;
+
+    public function __construct(
+        ArpGenerator $arpMetadataGenerator,
+        PrivacyQuestionsMetadataGenerator $privacyQuestionsMetadataGenerator,
+        MotivationMetadataGenerator $motivationMetadataGenerator,
+        SpDashboardMetadataGenerator $spDashboardMetadataGenerator
+    ) {
+        $this->arpMetadataGenerator = $arpMetadataGenerator;
+        $this->privacyQuestionsMetadataGenerator = $privacyQuestionsMetadataGenerator;
+        $this->motivationMetadataGenerator = $motivationMetadataGenerator;
+        $this->spDashboardMetadataGenerator = $spDashboardMetadataGenerator;
+    }
+
+    /**
+     * @param Entity $entity
+     * @return array
+     */
+    public function generate(Entity $entity)
+    {
+        $metadata = [
+            'active'          => true,
+            'allowedEntities' => [],
+            'allowedall'      => true,
+            'arp'             => $this->arpMetadataGenerator->build($entity),
+            'entityid'        => $entity->getEntityId(),
+            'metaDataFields'  => $this->generateMetadataFields($entity),
+            'metadataurl'     => $entity->getMetadataUrl(),
+            'state'           => 'testaccepted',
```

Comment:
Will the JSON attachement in the 'copy to prod' mail message have this state? This would be incorrect in my opinion.

Reply:
As discussed, this is desired behaviour.